app/controllers/samples_controller.rb

```diff
@@ -981,12 +982,18 @@ def stage_results
         end
       end
 
-      render json: {
-        pipeline_stage_results: {
-          pipeline_version: pipeline_run.pipeline_version,
-          stages: stage_info
-        }
+      @results = {
+        pipeline_version: pipeline_run.pipeline_version,
+        stages: stage_info
       }
+      respond_to do |format|
+        format.html do
```

Comment:
you could also just inline these like `format.html { render template: "samples/stage_results" }`